sup3r/preprocessing/batch_handling.py

```diff
@@ -542,6 +523,24 @@ def __init__(
         logger.info('Finished initializing BatchHandler.')
         log_mem(logger, log_level='INFO')
 
+    @property
+    def handler_weights(self):
+        """Get weights used to sample from different data handlers based on
+        relative sizes"""
+        sizes = [dh.size for dh in self.data_handlers]
+        weights = sizes / np.sum(sizes)
+        return weights
+
+    def get_handler_index(self):
+        """Get random handler index based on handler weights"""
+        indices = np.arange(0, len(self.data_handlers))
+        return np.random.choice(indices, p=self.handler_weights)
+
+    def get_current_handler(self):
```

Comment:
I think "get_rand_handler" would be more appropriate than "current". The latter sounds like something else needs to be run to randomize/select the handler.